plugins/source/azure/resources/services/nginx/deployments_fetch.go

```diff
@@ -0,0 +1,32 @@
+package nginx
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/nginx/armnginx"
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchDeployments(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+	// This is required so we don't override mock test options
+	if cl.Options == nil {
+		cl.Options = &arm.ClientOptions{}
+	}
+	cl.Options.APIVersion = "2022-11-01-preview"
```

Comment:
In the SDK... it sets the API version to be `2022-08-01`

https://github.com/Azure/azure-sdk-for-go/blob/a97519174f72085debd57f0b1189aaa2b62f5f7b/sdk/resourcemanager/nginx/armnginx/deployments_client.go#L287


I am wondering if this is an issue we should raise with the SDK team?

Reply:
OK so the the code you linked is for v2.0.0 and we're using `v1.0.1` I'll try to update first

Reply:
Yeah in `v1.0.1` it seems like they don't set the API version:
https://github.com/Azure/azure-sdk-for-go/blob/b11b29e0b4bf8ea6112ca3167574a4e7ee69a42a/sdk/resourcemanager/nginx/armnginx/deployments_client.go#LL272-L277C6

I assumed we were using the latest version of the SDK

Reply:
So updating to `v2` fixes it 🚀 I checked the other PRs and for those resources we're on the latest version.
There's a bigger issue here that our codegen doesn't support versions greater than `v1` (we'll need to fix that)